### DISCUSSION TITLE
CI (Continous Integration) / run Gradle tests for each PR?

### DISCUSSION THREAD

**Discussion by mccartney** [streetcomplete/StreetComplete#3957] [Repository Maintenance]
Would the project benefit if `./gradlew test` was run for all raised PRs?

As far as I can tell there's nothing like this set up.

**Comment by FloEdelmann** [comment]
@westnordost doesn't like that. See https://github.com/streetcomplete/StreetComplete/pull/3649#issuecomment-1018701994, https://github.com/streetcomplete/StreetComplete/pull/3760, https://github.com/streetcomplete/StreetComplete/pull/1917.

**Reply by mccartney** [reply]
If I read them correctly, all three pertain to running ktlint linter on each PR, while I suggest something (slightly) different - to run automated tests on each PR.

If it's about ktlint, I'd agree (as of now) for simple reason - ktlint for current `master` reports a lot of warnings, esp. deprecation-based. So every PR would have a lot of them (regardless if it introduced any or not).

**Reply by FloEdelmann** [reply]
I can't currently find the exact quote, but he said that it should be as easy as possible to contribute and failing (for whatever reason) tests might be scaring in a PR.

> ktlint for current `master` reports a lot of warnings, esp. deprecation-based. So every PR would have a lot of them (regardless if it introduced any or not).

That's not true; it's currently passing without any warnings (see [latest workflow run](https://github.com/streetcomplete/StreetComplete/runs/5971467694?check_suite_focus=true)). But I agree, there are often lint issues that are not *that* important to fix in a PR; successful compilation and passing tests would be more important.

**Reply by mccartney** [reply]
> I can't currently find the exact quote, but he said that it should be as easy as possible to contribute and failing tests (for whatever reason) might be scaring in a PR.

OK, gotcha.

**Reply by mccartney** [reply]
> That's not true; it's currently passing without any warnings

Sorry. You're right. I was confused by the output of `./gradlew lint` run locally. It printed a lot of warnings, but looking at them now maybe they are not from the linter itself, but just the compiler.

**Reply by FloEdelmann** [reply]
Note that `./gradlew lint` and `ktlint` are not the same thing.

**Reply by peternewman** [reply]
> I can't currently find the exact quote, but he said that it should be as easy as possible to contribute and failing (for whatever reason) tests might be scaring in a PR.

Although we could always make the tests passing be optional rather than mandatory which might make them less scary (if less effective).

For anyone else who finds this, in #3965 I added the ability to run them easily manually via an action at least.
